### DISCUSSION TITLE
Support modality for Menu

### DISCUSSION THREAD

**Discussion by Jonas-C** [chakra-ui/zag#1722] [Ideas]
# 🚀 Feature request

Being able to control the modality of a Menu would be great!

## 🧱 Problem Statement / Justification

In some Menu compositions, clicking outside of the Menu should not cause a click event to be triggered on the clicked element. 

## ✅ Proposed solution or API

The exact same API as Dialog and Popover. `Menu.Root modal`

## ↩️ Alternatives

You could probably find a way around this by utilizing either `onPointerDownOutside` or `onInteractOutside`. If this is deemed to be out of scope, I'll do that!

## 📝 Additional Information

https://www.radix-ui.com/primitives/docs/components/dropdown-menu

**Comment by segunadebayo** [comment]
Totally valid request. Want to give it a shot?

**Reply by segunadebayo** [reply]
Related: https://github.com/chakra-ui/ark/issues/2874

**Reply by dannylin108** [reply]
@segunadebayo  @Jonas-C  any update on this? 🙏
